### PR TITLE
Cherry-picker: Remove BACKPORT_COMPLETE. UNSET state after finished.

### PR DESCRIPTION
**Pull request by Mariatta**

### CODE REVIEW

**Comment by Mariatta** [comment]
@webknjaz @asvetlov can you take a look. miss-islington is seeing the "BACKPORT_COMPLETE" status, but couldn't continue because it is not an allowed state. I think it should be an allowed state, or did you have different expectation.

**Comment by webknjaz** [comment]
Checking it now.

**Comment by webknjaz** [comment]
@Mariatta I expected that state would be reset if everything is completed successfully. How about calling `reset_state()` after line 391 instead?

**Comment by webknjaz** [comment]
Any non-allowed states are supposed idicate that something went wrong.

**Comment by Mariatta** [comment]
Ok I've removed BACKPORT_COMPLETE. unset will be called instead, and using the UNSET state.

**Comment by Mariatta** [comment]
uugh don't know what's the deal with travis 😟 Tests-passing-locally-for-me™️

**Comment by webknjaz** [comment]
Oh, that's weird... Did you run all tests locally or just related?

**Comment by Mariatta** [comment]
Hmm I ran everything `pytest cherry_picker/test.py -v` in the cherry_picker directory

**Comment by webknjaz** [comment]
I checked this branch locally as well and it passes.

**Comment by webknjaz** [comment]
@Mariatta just to be on the safe side, could you please clear the cache in Travis CI? It's here https://travis-ci.org/python/core-workflow/caches

**Comment by Mariatta** [comment]
I've cleared cache and restart the build, still no luck 🤕 Anyone else can try this locally?

**Comment by webknjaz** [comment]
@Mariatta could you please merge #311 and then rebase this one? The test suite didn't use new fixtures in some tests and it could improve the situation, I think (hope).

**Comment by webknjaz** [comment]
@Mariatta I know what the problem is!

So `git log -r dc896437c8efe5a4a5dfa50218b7a6dc0cbe2598` is executed to check the repo.
But Travis CI only gets 50 last commits by default, not the whole tree. So this commit is now further than 50 commits and the check fails.

**Comment by webknjaz** [comment]
Solutions:
1) Change the sha in config to something newer
2) Find `&install-and-test-cherry-picker` on line 28 in `.travis.yml` and extend the mapping to contain Git config adjustment:
```yaml
- &install-and-test-cherry-picker
  ...
  git:
    depth: 200
  ...
```

File: cherry_picker/cherry_picker/cherry_picker.py [hunk 1]
```diff
@@ -16,19 +16,22 @@
...
-                          "{config[team]}/{config[repo]}/pulls")
-DEFAULT_CONFIG = collections.ChainMap({
-    'team': 'python',
-    'repo': 'cpython',
-    'check_sha': '7f777ed95a19224294949e1b4ce56bbffcb1fe9f',
-    'fix_commit_msg': True,
-    'default_branch': 'master',
-})
+CREATE_PR_URL_TEMPLATE = (
+    "https://api.github.com/repos/" "{config[team]}/{config[repo]}/pulls"
```
Comment (webknjaz) [inline cherry_picker/cherry_picker/cherry_picker.py]: I believe, one-line style doesn't require two chunks to be separated
```suggestion
    "https://api.github.com/repos/{config[team]}/{config[repo]}/pulls"
```

File: cherry_picker/cherry_picker/test.py [hunk 2]
```diff
@@ -39,216 +52,241 @@ def changedir(d):
...
+    [
+        (["3.1", "2.7", "3.10", "3.6"], ["3.10", "3.6", "3.1", "2.7"]),
+        (
+            ["stable-3.1", "lts-2.7", "3.10-other", "smth3.6else"],
+            ["3.10-other", "smth3.6else", "stable-3.1", "lts-2.7"],
+        ),
+    ],
+)
+@mock.patch("os.path.exists")
+@mock.patch("cherry_picker.cherry_picker.validate_sha")
```
Comment (webknjaz) [inline cherry_picker/cherry_picker/test.py]: Why?

Comment (Mariatta) [reply cherry_picker/cherry_picker/test.py]: These tests aren't testing validate_sha itself, but the other functionalities. I think it is ok to be mocked.

**Comment by Mariatta** [comment]
hm, but changing it to recent sha also only delays the problem, same issue will happen again in the future.

File: cherry_picker/cherry_picker/test.py (hunk 2 above)
Comment (webknjaz) [reply cherry_picker/cherry_picker/test.py]: I disagree: more integrated tests is better. Testing a combination of things makes it all more robust. Especially when it costs nothing.
As in

**Comment by webknjaz** [comment]
@Mariatta I think you should just make git clone deeper. Or if you're worried about the delay, it's possible to disable `--depth=` in Travis CI so that it'd clone the whole thing.

**Comment by Mariatta** [comment]
Thanks, I've disabled git depth in travis CI and reverted the mocks.

**Comment by webknjaz** [comment]
@Mariatta great!

P.S. One small feedback about `black out`: as you can see it messes up unrelated stuff making it impossible to track down (by looking at diff) what was the intended change in the PR. It's not completely non-atomic. I'd recommend applying black to the project separately and then when PR is on top of that, using back will only generate related changes :)

**Comment by Mariatta** [comment]
Thanks!

**Pull request merged**